drivers/include/lm75.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#ifndef LM75_H
+#define LM75_H
+
+#include "periph/i2c.h"
+
+/**
+ * @name LM75 return values
+ */
+enum {
+    LM75_SUCCESS = 0,
+    LM75_ERROR_I2C,
+    LM75_ERROR
+};
+
+/* temperature resolution of the LM75 sensors */
+typedef struct {
+    uint16_t    os_res;
+    uint8_t     os_bits;
+    uint16_t    os_mult;
+    uint16_t    temp_res;
+    uint8_t     temp_bits;
+    uint16_t    temp_mult;
```

Comment:
btw: you can eliminate padding by ordering the struct elements by size, descending

http://www.catb.org/esr/structure-packing/